File: frontend/src/types/seller.ts
import type { UserPublic } from './user'

export interface SellerProfile {
  id: string
  user: UserPublic
  headline: string | null
  bio: string | null
  category_id: string | null
  career_years: number | null
  review_count: number
  average_rating: number | null
  created_at: string
}

export interface SellerProfileInput {
  headline?: string | null
  bio?: string | null
  category_id?: string | null
  career_years?: number | null
}

export type PortfolioStatus = 'draft' | 'published'

export interface PortfolioPost {
  id: string
  seller_profile_id: string
  title: string
  content: string
  status: PortfolioStatus
  created_at: string
  updated_at: string
}

export interface PortfolioPostSummary {
  id: string
  title: string
  status: PortfolioStatus
  thumbnail: string | null
  created_at: string
}

export interface PortfolioPostInput {
  title: string
  content: string
  status: PortfolioStatus
}

File: backend/app/schemas/seller.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import PortfolioStatus
from app.schemas.common import ORMBase
from app.schemas.user import UserPublic


class SellerProfileCreate(BaseModel):
    headline: str | None = Field(default=None, max_length=255)
    bio: str | None = None
    category_id: uuid.UUID | None = None
    career_years: int | None = None


class SellerProfileUpdate(SellerProfileCreate):
    pass


class SellerProfileOut(ORMBase):
    id: uuid.UUID
    user: UserPublic
    headline: str | None = None
    bio: str | None = None
    category_id: uuid.UUID | None = None
    career_years: int | None = None
    review_count: int = 0
    average_rating: float | None = None
    created_at: datetime


class PortfolioPostCreate(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    content: str
    status: PortfolioStatus = PortfolioStatus.draft


class PortfolioPostUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=255)
    content: str | None = None
    status: PortfolioStatus | None = None


class PortfolioPostOut(ORMBase):
    id: uuid.UUID
    seller_profile_id: uuid.UUID
    title: str
    content: str
    status: PortfolioStatus
    created_at: datetime
    updated_at: datetime


class PortfolioPostSummary(ORMBase):
    id: uuid.UUID
    title: str
    status: PortfolioStatus
    thumbnail: str | None = None
    created_at: datetime
